tribe-ext-ea-additional-options.php

```diff
@@ -0,0 +1,324 @@
+                        ></span>
+                </div>
+                <div class="tribe-refine tribe-active">
+                    <label for="tribe-ea-field-prefix"><?php esc_html_e('Event Title Prefix:', 'tribe-ext-ea-additional-options'); ?></label>
+                    <input id="tribe-ea-field-prefix" name="aggregator[prefix]" class="tribe-ea-field tribe-ea-size-large" type="text" value="<?php echo esc_attr($prefixValue); ?>" />
+                    <span
+                        class="tribe-bumpdown-trigger tribe-bumpdown-permanent tribe-bumpdown-nohover tribe-ea-help dashicons dashicons-editor-help"
+                        data-bumpdown="<?php echo esc_attr__('Add text before the title of each event.', 'tribe-ext-ea-additional-options'); ?>"
+                        data-width-rule="all-triggers"
+                        ></span>
+                </div>
+            </div>
+            <?php
+        }
+
+        /**
+         * Checks website url setting and filters link
+         * 
+         * @param string $link
+         * @param int $postId
+         * @return string
+         */
+        public function filter_event_link($link, $postId) {
+            $website_url = tribe_get_event_website_url($postId);
+            if (!empty($website_url)) {
+                return $website_url;
+            }
+            return $link;
+        }
+
+        public function filter_service_data_field_map($fieldMap) {
+            if (isset($fieldMap['description'])) {
+                unset($fieldMap['description']);
+            }
+            $fieldMap['unsafe_description'] = 'post_content';
+            return $fieldMap;
+        }
+
+        /**
+         * Filters event info before being saved or updated
+         * 
+         * @param array $event
+         * @return array
+         */
+        public function filter_imported_event($event, $record) {
+            $meta = $record->meta;
+            $lineBreakOpt = tribe_get_option($this->opts_prefix . 'retain_line_breaks');
+            if (tribe_is_truthy($lineBreakOpt)) {
+                $event['post_content'] = str_replace(['\\n', '\n', "\n", "\\n"], '<br>', $event['post_content']);
+            }
+            if (!empty($meta['prefix'])) {
+                $event['post_title'] = $meta['prefix'] . ' ' . $event['post_title'];
+            }
+            if (!empty($meta['timezone'])) {
+                if (tribe_is_truthy($event['EventAllDay'])) {
+                    $event['EventTimezone'] = $meta['timezone'];
+                } else {
+                    $utcTimezone = new DateTimeZone("UTC");
+                    $targetOffset = timezone_offset_get(timezone_open($meta['timezone']), new DateTime('now', $utcTimezone));
+                    $eventOffset = timezone_offset_get(timezone_open($event['EventTimezone']), new DateTime('now', $utcTimezone));
+                    $currTimezone = new DateTimeZone($event['EventTimezone']);
```

Comment:
Forgot to mentioned that `DateTimeZone` and `DateTime` triggers exceptions in case the input of the constructor is not valid in this case you can catch this errors as follows.

```suggestion
                    try { 
                         $currTimezone = new DateTimeZone($event['EventTimezone']);
                    } catch ( \Exception $e ) {
                        $currTimezone = $utcTimezone;
                    }
```